README.md

```diff
@@ -88,9 +92,35 @@ KillTimer(d time.Duration, errCh chan<- error)
 KillAfter(t time.Time, errCh chan<- error)
 ```
 
+The standard library
+function [signal.NotifyContext](https://cs.opensource.google/go/go/+/go1.17.1:src/os/signal/signal.go;l=277) is one
+provides a convenient contextual hook over interrupts. Remember that putting a hook on an interrupt will intercept it,
+and can prevent exit if you don't take manual steps to gracefully shut down.
+
+```go
+    ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
+cmd := NewWithContext(ctx, "ls")
+// etc…
+```
+
+## Impersonating a user
+
+You can impersonate a valid user by using the `Impersonate(username string) error` function:
+
+```go
+cmd := New("ls", "/")
+_ = cmd.Impersonate("username", false)
+
+out, _ := cmd.CombinedOutput()
+fmt.Println(out)
+```
+
+Your user needs to be capable of assuming the target user otherwise you'll receive an `operation not permitted` error.
```

Comment:
👍 good to know this was one of my questions

Reply:
Yeah, it was something I struggled against in the IDE, then at command line, then I thought `sudo` bit needed, and found a way of isolating the test and running it once that was sorted.